packages/mongodb-build-info/src/index.ts

```diff
@@ -134,6 +140,41 @@ export function getGenuineMongoDB(uri: string): {
   };
 }
 
+export async function identifyServerName(
+  uri: string,
+  runCommand: (command: Document) => Promise<Document>,
+  adminCommand: (command: Document) => Promise<Document>,
+): Promise<string> {
+  const hostname = getHostnameFromUrl(uri);
+  if (hostname.match(COSMOS_DB_REGEX)) {
+    return 'cosmosdb';
+  }
+
+  if (hostname.match(DOCUMENT_DB_REGEX)) {
+    return 'documentdb';
+  }
+
+  if (hostname.match(FIRESTORE_REGEX)) {
+    return 'firestore';
+  }
+
+  const buildInfo = await runCommand({ buildInfo: 1 });
```

Comment:
I think we also need to account for the possibility of this command failing (either because we're unauthenticated or because we're using the versioned/stable API feature of the server)